### ISSUE TITLE
python_requires is >= 2.7 but codebase uses f-strings

### ISSUE THREAD

**Issue by thatlittleboy** [holoviz/holoviews#5276]
#### ALL software version info
holoviews latest release v1.14.8
Python 3.5.3

Other software versions are irrelevant to the bug report.

#### Description of expected behavior and the observed behavior

<img src="https://user-images.githubusercontent.com/30731072/164188402-5ac973c0-cdd0-4ea5-98ae-97dedf8fbb67.png" width=400>

The README touts that python 3.6+ is supported, and indeed, the codebase is making use of f-strings (for example), making the code base incompatible with older versions (<3.6) of Python.

However, the `setup.py` [here](https://github.com/holoviz/holoviews/blob/c68e330acc8bf3e2da07e79c5eed6c6aa2d121f6/setup.py#L141) is still set-up with py2.7+.

This means that systems in py<3.6 are still pip installing the latest holoviews version by default. Which cannot run, because of the aforementioned code incompatibilities.

#### Complete, minimal, self-contained example code that reproduces the issue
NIL. `pip install holoviews` on a python 3.5 system. The latest version is installed by default. Then import holoviews.

#### Stack traceback and/or browser JavaScript console output

import holoviews gives this:

```
Traceback (most recent call last):

  File "/xxxx/venv/lib/python3.5/site-packages/IPython/core/interactiveshell.py", line 3326, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)

  File "<ipython-input-123-3c97b9bb50ff>", line 1, in <module>
    import holoviews

  File "/xxxx/venv/lib/python3.5/site-packages/holoviews/__init__.py", line 11, in <module>
    from . import util                                       # noqa (API import)

  File "/xxxx/venv/lib/python3.5/site-packages/holoviews/util/__init__.py", line 15, in <module>
    from ..core import (

  File "/xxxx/venv/lib/python3.5/site-packages/holoviews/core/__init__.py", line 4, in <module>
    from .data import *            # noqa (API import)

  File "/xxxx/venv/lib/python3.5/site-packages/holoviews/core/data/__init__.py", line 25, in <module>
    from ..element import Element

  File "/xxxx/venv/lib/python3.5/site-packages/holoviews/core/element.py", line 7, in <module>
    from .layout import Composable, Layout, NdLayout

  File "/xxxx/venv/lib/python3.5/site-packages/holoviews/core/layout.py", line 24
    raise TypeError(f"unsupported operand type(s) for +: {x.__class__.__name__} and {y.__class__.__name__}. "
                                                                                                            ^
SyntaxError: invalid syntax
```

-----

**EDIT:** Seems like I had it backwards. I just read the release notes of 1.14.0 that 1.14.0 is the last release supporting py2.7, which means the f-strings were wrongly added in an earlier PR.

**Comment by philippjfr** [comment]
Looks like this was accidentally backported onto the 1.14.x branch. Should fix this in 1.14.9.

**Comment by thatlittleboy** [comment]
@philippjfr Is this fixed in 1.14.9? I noticed it was released already but no mention of this issue in the release notes?

**Comment by maximlt** [comment]
@thatlittleboy usually the release notes mention the PRs and not the issues. Version 1.14.9 re-established compatibility with Python 2, at least that's what we tried to do! As indeed, setting up the test suite on Python 2 has just proven to be impossible, the world has changed too much since that era and in particular the PyData world. However I believe some manual tests were run to check that the compatibility with Py2 was effectively achieved.

 I am going to close this issue, feel free to add a comment if Version 1.14.9 did not achieve the Python 2 compatibility.
